feat: add output.minify config (#1730)
web-infra-dev/rsbuild: packages/document/docs/en/config/output/minify.mdx

# output.minify

- **Type:**

```ts
export type Minify =
  | boolean
  | {
      js?: boolean;
      jsOptions?: SwcJsMinimizerRspackPluginOptions;
      css?: boolean;
      html?: boolean;
      htmlOptions?: HTMLPluginOptions['minify'];
    };
```

- **Default:** `true`

> Configure whether to enable and configure code minification in the production environment.

Reviewer: ```suggestion
Configure whether to enable code minification in the production build, or to configure minimiser options.
```

By default, JS, CSS, and HTML code will be automatically minimized during production environment builds to enhance page performance. If you do not wish to perform code minification, you can set `minify` to `false` to turn off minification for all code. Alternatively, you can control the behavior of code minification through detailed configuration of the `minify` option. Below are detailed explanations for each configuration item:

Here are explanations for each field:

- `js`: Whether to enable minification for JavaScript code.
- `jsOptions`: JS code minification configuration, which will be merged with the default configuration and passed to SWC.
- `css`: Whether to enable minification for CSS code.
- `html`: Whether to enable minification for HTML code.
- `htmlOptions`: HTML code minification configuration, which will be merged with the default configuration and passed to html-rspack-plugin.

## Example

### Disable all code minification

```js
export default {
  output: {
    minify: false,
  },
};
```

### Disable JavaScript code minification

```js
export default {
  output: {
    minify: {
      js: false,
    },
  },
};
```

> ### Customize JavaScript code minification

Reviewer: We can add an example for `Customize HTML minification` too.
